pkg/reconciler/workload/synctarget/synctarget_reconcile.go

```diff
@@ -34,6 +34,13 @@ import (
 func (c *Controller) reconcile(syncTarget *workloadv1alpha1.SyncTarget, workspaceShards []*v1alpha1.ClusterWorkspaceShard) (*workloadv1alpha1.SyncTarget, error) {
 	syncTargetCopy := syncTarget.DeepCopy()
 
+	labels := syncTargetCopy.GetLabels()
```

Comment:
you can edit `syncTargetCopy.Labels` directly. No need for Get/SetLabel.

Reply:
that panics.

Reply:
no. Of course you have to create the map if it is nil. But no need for calling setter and getter.